app/Commands/Dev/Core/Compile/Base.hs

```diff
@@ -24,7 +24,10 @@ getEntry PipelineArg {..} = do
     ep
       { _entryPointTarget = getTarget (_pipelineArgOptions ^. compileTarget),
         _entryPointDebug = _pipelineArgOptions ^. compileDebug,
-        _entryPointOptimizationLevel = _pipelineArgOptions ^. compileOptimizationLevel,
+        _entryPointOptimizationLevel =
+          if
+              | _pipelineArgOptions ^. compileDebug -> 0
```

Comment:
This has the disadvantage that one cannot override the optimization level with `-On` when compiling in debug mode. The `compileOptimizationLevel` should probably be a `Maybe`, defaulting to 0 or 1 if it's `Nothing` depending on whether `--debug` is enabled.